### PR TITLE
Support Xec

### PR DESCRIPTION
**Pull request by KenNguyen-dev**

### CODE REVIEW

File: apps/api/src/wallet/xpi-wallet.service.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        masterHDNode,
+        path: path
+      });
+      walletPaths.push(walletPath);
+    }
+
+    return walletPaths;
+  };
+
+  async deriveAccount (masterHDNode)  {
```
Comment (vince8x) [inline apps/api/src/wallet/xpi-wallet.service.ts:71]: This is duplicated of xec's deriveAccount. Can we move the method to the parent class?

File: apps/api/src/wallet/xec-wallet.service.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  
+      const broadcastResponse = await this.chronik.broadcastTx(rawTxHex).catch(async err => {
+        const updatingWalletFund = 'Updating wallet fund';
+        throw new Error(updatingWalletFund);
+      });
+      const { txid } = broadcastResponse; 
+  
+      return txid;
+    }catch(e) {
+      console.log(e);
```
Comment (vince8x) [inline apps/api/src/wallet/xec-wallet.service.ts:158]: We can use logger instead of console here

File: apps/api/src/wallet/xpi-wallet.service.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  
+      const broadcastResponse = await this.chronik.broadcastTx(rawTxHex).catch(async err => {
+        const updatingWalletFund = 'Updating wallet fund';
+        throw new Error(updatingWalletFund);
+      });
+      const { txid } = broadcastResponse; 
+  
+      return txid;
+    }catch(e) {
+      console.log(e);
```
Comment (vince8x) [inline apps/api/src/wallet/xpi-wallet.service.ts:164]: We can use logger instead of console

File: apps/api/src/utils/useXEC.ts [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+      // Throw validation error if fromXecToSatoshis returns false
+      if (!satoshisToSend) {
+          const error = new Error(`Invalid decimal places for send amount`);
+          throw error;
+      }
+
+      let encryptedEj; // serialized encryption data object
+
+      // if the user has opted to encrypt this message
```
Comment (vince8x) [inline apps/api/src/utils/useXEC.ts:135]: Please remove unused code

File: apps/api/src/gitbot/account.service.ts [hunk 5]
```diff
@@ -112,6 +114,150 @@ export class AccountService {
...
+                    ...xecWalletPath,
+                    coin: Coin.XEC
+                  }
+                ]
+              }
+            }
+          }
+        })
+
+        // await this.prisma.walletPath.create({
```
Comment (vince8x) [inline apps/api/src/gitbot/account.service.ts:208]: Please remove unused code

File: apps/api/src/gitbot/account.service.ts [hunk 6]
```diff
@@ -112,6 +114,150 @@ export class AccountService {
       throw new Error(`createAccount: ${e.message}`);
     }
   };
+
+  /**
+   * - Save platformId/user/account to database
+   * - Return `accountId` and `userId` from saved account
+   */
+  public createAccountTest = async (
```
Comment (vince8x) [inline apps/api/src/gitbot/account.service.ts:122]: Please add "@todo" to remove later

File: apps/api/src/utils/cashMethods.ts [hunk 7]
```diff
@@ -543,3 +614,236 @@ export const getAddressesOfWallet = wallet => {
...
+  return amountInBaseUnits;
+};
+
+export const sumOneToManyXec = destinationAddressAndValueArray => {
+  return destinationAddressAndValueArray.reduce((prev, curr) => {
+      return parseFloat(prev) + parseFloat(curr.split(',')[1]);
+  }, 0);
+};
+
+export const generateXecTxInput = (
```
Comment (vince8x) [inline apps/api/src/utils/cashMethods.ts:672]: Please add type signature as much as possible

File: apps/api/src/utils/cashMethods.ts [hunk 8]
```diff
@@ -543,3 +614,236 @@ export const getAddressesOfWallet = wallet => {
...
+      const tx = txBuilder.build();
+      // output rawhex
+      hex = tx.toHex();
+  } catch (err) {
+      throw new Error('Transaction build failed');
+  }
+  return hex;
+};
+
+export const generateXecTxOutput = (
```
Comment (vince8x) [inline apps/api/src/utils/cashMethods.ts:781]: Please add type signature

File: apps/api/src/utils/useXEC.ts [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+          }
+      }
+  }
+  // You get here if you find no outgoing txs in the chronik tx history
+  throw new Error(
+      'Cannot send an encrypted message to a wallet with no outgoing transactions in the last 20 txs',
+  );
+};
+
+export const sendXec = async (
```
Comment (vince8x) [inline apps/api/src/utils/useXEC.ts:91]: Please add type signature

File: apps/api/src/utils/useXEC.ts (hunk 4 above)
Comment (KenNguyen-dev) [reply apps/api/src/utils/useXEC.ts:135]: We can leave it here in case we need encrypted message

**Review by vince8x — APPROVED** [review]

**Pull request merged**